Build failure: whisper-cpp (dependened by ffmpeg/kdenlive) with rocmSupport

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

nixpkgs.config.rocmSupport = true;

install ffmpeg-full/kdenlive

### Can Hydra reproduce this build failure?

Please select the Hydra Status.

### Link to Hydra build job

no

### Relevant log output

```console
running for 10hours. tried again, no output after this:

whisper-cpp>        |                       operator""_json                                                                                                                                                        
whisper-cpp> /build/source/examples/json.hpp:24322:49: warning: identifier '_json_pointer' preceded by whitespace in a literal operator declaration is deprecated [-Wdeprecated-literal-operator]                  
whisper-cpp>  24322 | inline nlohmann::json::json_pointer operator "" _json_pointer(const char* s, std::size_t n)                                                                                                  
whisper-cpp>        |                                     ~~~~~~~~~~~~^~~~~~~~~~~~~                                                                                                                                
whisper-cpp>        |                                     operator""_json_pointer                                                                                                                                  
whisper-cpp> /build/source/examples/json.hpp:24386:58: warning: identifier '_json' preceded by whitespace in a literal operator declaration is deprecated [-Wdeprecated-literal-operator]                          
whisper-cpp>  24386 |     using nlohmann::literals::json_literals::operator "" _json; // NOLINT(misc-unused-using-decls,google-global-names-in-headers)                                                            
whisper-cpp>        |                                              ~~~~~~~~~~~~^~~~~                                                                                                                               
whisper-cpp>        |                                              operator""_json                                                                                                                                 
whisper-cpp> /build/source/examples/json.hpp:24387:58: warning: identifier '_json_pointer' preceded by whitespace in a literal operator declaration is deprecated [-Wdeprecated-literal-operator]                  
whisper-cpp>  24387 |     using nlohmann::literals::json_literals::operator "" _json_pointer; //NOLINT(misc-unused-using-decls,google-global-names-in-headers)                                                     
whisper-cpp>        |                                              ~~~~~~~~~~~~^~~~~~~~~~~~~                                                                                                                       
whisper-cpp>        |                                              operator""_json_pointer                                                                                                                         
whisper-cpp> 4 warnings generated when compiling for host.
```

### Additional context

last working nixpkgs: 80bdc1e5ce51f56b19791b52b2901187931f5353 first broken: aca4d95fce4914b3892661bcb80b8087293536c6

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.19.6-cachyos, NixOS, 26.05 (Yarara), 26.05.19700101.dirty`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.3`
 - nixpkgs: `/nix/store/i5i1xs938x4nkafclp1riwxpn7g0z3fj-nixpkgs-patched`



### Notify maintainers

@dit7ya @hughobrien @aviallon

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/rocm-modules/llvm/default.nix
+++ b/pkgs/development/rocm-modules/llvm/default.nix
@@ -306,6 +306,29 @@ overrideLlvmPackagesRocm (s: {
         stripLen = 1;
       })
       ./perf-increase-namestring-size.patch
+      # v64i8 shuffle lowering inf loop on VBMI targets, hangs whisper-cpp etc
+      # https://github.com/NixOS/nixpkgs/issues/497745
+      (fetchpatch {
+        # https://github.com/llvm/llvm-project/pull/182832
+        name = "llvm-x86-v64i8-add-test-coverage.patch";
+        url = "https://github.com/llvm/llvm-project/commit/0e3a96d0ec01e3575674d72c4e23bf98affdca28.patch";
+        relative = "llvm";
+        hash = "sha256-qhRkB8Fjz/fNacuGv1OFkiTNOQ0/QQ9p4pLFudwrTzM=";
+      })
+      (fetchpatch {
+        # https://github.com/llvm/llvm-project/pull/182852
+        name = "llvm-x86-v64i8-prefer-vpermv3-on-vbmi.patch";
+        url = "https://github.com/llvm/llvm-project/commit/8f5880d3ae4e5dfc748985d90e5413671028aa3e.patch";
+        relative = "llvm";
+        hash = "sha256-4DU6gu/1+iQpzvVYBlTTUKtw77QSRyTja4hdel4D5Cw=";
+      })
+      (fetchpatch {
+        # https://github.com/llvm/llvm-project/pull/183109
+        name = "llvm-x86-v64i8-skip-repeated-mask-lane-permute-on-vbmi.patch";
+        url = "https://github.com/llvm/llvm-project/commit/1b9fea021840f17c41ea980300d0fc45e7285909.patch";
+        relative = "llvm";
+        hash = "sha256-9Akm78QQr8BIMrVWwDG3poWS1HuQ0hpIQWfke3oADgg=";
+      })
       # TODO: consider reapplying "Don't include aliases in RegisterClassInfo::IgnoreCSRForAllocOrder"
       # it was reverted as it's a pessimization for non-GPU archs, but this compiler
       # is used mostly for amdgpu

